tests/fuzz/fuzz-amount.c

```diff
@@ -0,0 +1,62 @@
+#include <assert.h>
+#include <stdint.h>
+#include <tests/fuzz/libfuzz.h>
+
+#include <ccan/tal/tal.h>
+#include <common/amount.h>
+#include <common/utils.h>
+
+void init(int *argc, char ***argv)
+{
+}
+
+void run(const uint8_t *data, size_t size)
+{
+	struct amount_msat msat;
+	struct amount_sat sat;
+	char *string;
+	uint8_t *buf;
+	const char *fmt_msat, *fmt_msatbtc, *fmt_sat, *fmt_satbtc;
+
+
+	/* We should not crash when parsing any string. */
+
+	string = tal_arr(NULL, char, 0);
+	for (size_t i = 0; i < size; i++)
+		if (data[i] <= CHAR_MAX)
+			tal_arr_expand(&string, (char)data[i]);
```

Comment:
Nice catch, thanks, applied.

```patch
commit e3fdb759be3867416cbfb62ef18b8a23f2c6b930 (HEAD -> fuzzing)
Author: Antoine Poinsot <darosior@protonmail.com>
Date:   Mon Oct 19 16:02:55 2020 +0200

    fixup! fuzz: add a fuzz target for amounts parsing
    
    Signed-off-by: Antoine Poinsot <darosior@protonmail.com>

diff --git a/tests/fuzz/fuzz-amount.c b/tests/fuzz/fuzz-amount.c
index b82e5b970..644c9b0b7 100644
--- a/tests/fuzz/fuzz-amount.c
+++ b/tests/fuzz/fuzz-amount.c
@@ -21,10 +21,9 @@ void run(const uint8_t *data, size_t size)
 
        /* We should not crash when parsing any string. */
 
-       string = tal_arr(NULL, char, 0);
+       string = tal_arr(NULL, char, size);
        for (size_t i = 0; i < size; i++)
-               if (data[i] <= CHAR_MAX)
-                       tal_arr_expand(&string, (char)data[i]);
+               string[i] = (char) data[i] % (CHAR_MAX + 1);
 
        parse_amount_msat(&msat, string, tal_count(string));
        parse_amount_sat(&sat, string, tal_count(string));
```

I discarded pre-allocation for this because of my tests with `get_chunks`..